src/client/Client.h

```diff
@@ -934,8 +934,9 @@ class Client : public Dispatcher, public md_config_obs_t {
   void dump_mds_sessions(Formatter *f, bool cap_dump=false);
 
   int make_request(MetaRequest *req, const UserPerm& perms,
-		   InodeRef *ptarget = 0, bool *pcreated = 0,
-		   mds_rank_t use_mds=-1, bufferlist *pdirbl=0);
+                   InodeRef *ptarget = 0, bool *pcreated = 0,
+                   mds_rank_t use_mds=-1, bufferlist *pdirbl=0,
+                   size_t feature_needed=-1);
```

Comment:
```suggestion
                   size_t feature_needed=0);
```

assigning a negative value to an unsigned is weird.

Reply:
The `-1` means a `0xFFFFFFFFFFFFFFFF` for unsigned `size_t`. The `feature_bitset_t::test()` need a `size_t` parameter.

https://github.com/ceph/ceph/blob/main/src/mds/mdstypes.h#L315

Reply:
Fixed it by switching to use `ULONG_MAX` instead of `-1`.